storagemarket/impl/client.go

```diff
@@ -199,9 +200,21 @@ func (c *Client) ProposeStorageDeal(
 	collateral abi.TokenAmount,
 	rt abi.RegisteredProof,
 ) (*storagemarket.ProposeStorageDealResult, error) {
-	commP, pieceSize, err := clientutils.CommP(ctx, c.pio, rt, data)
-	if err != nil {
-		return nil, xerrors.Errorf("computing commP failed: %w", err)
+
+	var commP cid.Cid
+	var pieceSize abi.UnpaddedPieceSize
+	var err error
+
+	if data.TransferType != storagemarket.TTManual {
```

Comment:
the clientutils.CommP function already checks if its a manual transfer and short circuits. Maybe move this logic down into that?

Reply:
yea that makes sense